platform/ios/app/MBXEmbeddedMapViewController.m

```diff
@@ -0,0 +1,146 @@
+#import "MBXEmbeddedMapViewController.h"
+#import <Mapbox/Mapbox.h>
+
+typedef NS_ENUM(NSInteger, MBXEmbeddedControl) {
+    MBXEmbeddedControlZoom = 0,
+    MBXEmbeddedControlScroll,
+    MBXEmbeddedControlRotation,
+    MBXEmbeddedControlPitch
+};
+
+
+@interface MBXEmbeddedMapViewController () <UIScrollViewDelegate>
+
+@property (nonatomic, strong) MGLMapView *mapView;
+
+@end
+
+@implementation MBXEmbeddedMapViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    UIRotationGestureRecognizer *gestureRecognizer = [[UIRotationGestureRecognizer alloc] initWithTarget:self action:@selector(rotation:)];
+    
+    UIScrollView *scrollView = [[UIScrollView alloc] initWithFrame:self.view.frame];
+    scrollView.delegate = self;
+    [self.view addSubview:scrollView];
+    [scrollView addGestureRecognizer:gestureRecognizer];
+    
+    CGRect mapRect = CGRectMake(0, 0, CGRectGetWidth(scrollView.frame) * 2, CGRectGetHeight(scrollView.frame) * 2);
+    self.mapView = [[MGLMapView alloc] initWithFrame:mapRect];
+    [scrollView addSubview:self.mapView];
+    scrollView.contentSize = CGSizeMake(CGRectGetWidth(mapRect), CGRectGetHeight(mapRect));
+    
+    CGFloat widthScale = self.view.bounds.size.width / CGRectGetWidth(mapRect);
+    CGFloat heightScale = self.view.bounds.size.height / CGRectGetHeight(mapRect);
+    CGFloat minScale = MIN(widthScale, heightScale);
+    
+    scrollView.maximumZoomScale = minScale*10;
+    scrollView.minimumZoomScale = minScale;
+    
+    //Create list of all the possible control titles
+    NSMutableArray <UIView*>*items = [[NSMutableArray alloc] init];
+    for(NSInteger index = 0; index < 4; index++) {
+        UIView *view = [[UIView alloc] initWithFrame:CGRectZero];
+        view.backgroundColor = [UIColor colorWithWhite:1 alpha:0.5];
+        UILabel *label = [[UILabel alloc] initWithFrame:CGRectZero];
+        label.font = [UIFont systemFontOfSize:12];
+        label.translatesAutoresizingMaskIntoConstraints = NO;
+        label.text = [[self class] titleForControl:index];
+        UISwitch *switchControl = [[UISwitch alloc] initWithFrame:CGRectZero];
+        switchControl.tag = index;
+        [switchControl addTarget:self action:@selector(didSwitch:) forControlEvents:UIControlEventValueChanged];
+        switchControl.translatesAutoresizingMaskIntoConstraints = NO;
+        switchControl.on = [self statusForControl:index];
```

Comment:
The body of this `for` loop seems like a good candidate to break out into a separate method. Considering that there are only four switch controls in all, perhaps it makes sense to set each switch’s title explicitly, track them with separate properties, and hook them up to separate action methods. Someone theoretically adding a “Roll Enabled” switch (!) would have fewer methods to update.